blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DAbstractMetricIO.java

```diff
@@ -254,4 +254,14 @@ protected void addRollupToBatch(BatchStatement batch, Locator locator,
         }
         return locatorTimestampRollup;
     }
+
+    /**
+     * Retrieves the {@link BoundStatement} for a particular metric and granularity.
+     * Subclasses will implement this.
+     *
+     * @param metric
+     * @param granularity
+     * @return
+     */
+    protected abstract BoundStatement getBoundStatementForMetric(IMetric metric, Granularity granularity);
```

Comment:
Each of the subclasses override an abstract method called `getRollupType()` which returns (BluefloodSetRollup, BasicRollup, etc). The `getBoundStatementForMetric` calls `getRollupType()` to get the type. I dont know if that covers all the issues.

If its too much hastle, I am fine with leaving the way it is now.